# HiFi Metagenomics Publications

(Updated November 27, 2023)

## Pre-prints 

+ Benoit G, Raguideau S, James R, Phillippy AM, Chikhi R, and C Quince. 2023. Efficient high-quality metagenome assembly from long accurate reads using minimizer-space de Bruijn graphs. bioRxiv, https://doi.org/10.1101/2023.07.07.548136

+ Eisenhofer R, Nesme J, Santos-Bay L, Koziol A, Sorenson SJ, Alberdi A, and O Aizpurua. 2023. A comparison of short-read, HiFi long-read and hybrid strategies for genome-resolved metagenomics. bioRxiv, https://doi.org/10.1101/2023.10.04.560907 

+ Chen W-H, Chen J, Sun C, Dong Y, Jin M, Lai S, Jia L, Zhao X, Gao NL, Liu Z, Bork P, and X-M Zhao. 2022. Efficient recovery of complete gut phage genomes by combined short- and long-sequencing. bioRxiv, https://doi.org/10.1101/2022.07.03.498593

+ Kutuzova S, Nielsen M, Piera P, Nissen JN, and S Rasmussen. 2023. Taxometer: improving taxonomic classification of metagenomic contigs. bioRxiv, https://doi.org/10.1101/2023.11.23.568413

+ Masuda S, Gan P, Kiguchi Y, Anda M, Sasaki K, Shibata A, Iwasaki W, Suda W, and K Shirasu. 2023. Uncovering plant microbiomes using long-read metagenomic sequencing. bioRxiv, https://doi.org/10.1101/2023.03.20.533568

+ Priest T, Vidal-Melgosa S, Hehemann J-H, Amann R, and BM Fuchs. 2023. Spatial heterogeneity in carbohydrates and their utilisation by microbes in the high North Atlantic. bioRxiv, https://doi.org/10.1101/2023.05.11.540373

+ Shaw J, and YW Yu. 2023. Metagenome profiling and containment estimation through abundance-corrected k-mer sketching with sylph. bioRxiv, https://doi.org/10.1101/2023.11.20.567879

+ Valentin-Alvarado LE, Appler KE, De Anda V, Schoelmerich MC, West-Roberts J, Kivenson V, Crits-Christoph A, Ly L, Sachdeva R, Savage DF, Baker BJ, and JF Banfield. 2023. Asgard archaea modulate potential methanogenesis substrates in wetland soil. bioRxiv, https://doi.org/10.1101/2023.11.21.568159

+ Wang G, Zhao L, Shi Y, Qu F, Ding Y, Liu W, Liu C, Luo G, Li M, Bai X, Li L, Ho Y-P, and J Yu. 2023. High-throughput generic single-entity sequencing using droplet microfluidics. bioRxiv, https://doi.org/10.1101/2023.08.15.549386

+ Wang Y-C, Mao Y, Fu H-M, Wang F, Weng X, Liu Z-H, Xu X-W, Yan P, Fang F, Guo J-S, Shen Y, and Y-P Chen. 2023. Complete genome-centric analysis reveals functional divergence and adaptive evolution of uncultured bacteria in anammox community. bioRxiv, https://doi.org/10.1101/2023.08.15.553441

+ Yu W, Luo H, Yang J, Zhang S, Jiang H, Zhao X, Hui X, Sun D, Li L, Wei X-Q, Lonardi S, and W Pan. 2023. Comprehensive assessment of eleven de novo HiFi assemblers on complex eukaryotic genomes and metagenomes. bioRxiv, https://doi.org/10.1101/2023.06.29.546998


## Online Early

+ Jang YJ, Kang J-S, and EH Bae. 2023. Metagenome-assembled genomes of the GU0601 sample (the Han River, South Korea). Microbiology Resource Announcements, Online Early. https://doi.org/10.1128/MRA.00688-23

+ Sun C, Chen J, Jin M, Zhao X, Li Y, Dong Y, Gao N, Liu Z, Bork P, Zhao X-M, and W-H Chen. 2023. Long-read sequencing reveals extensive DNA methylations in human gut pangenome contributed by prevalently phage-encoded methyltransferases. Advanced Science, Online Early. https://doi.org/10.1002/advs.202302159



## 2023

+ Ahmad N, Ritz M, Calchera A, Otte J, Schmitt I, Brueck T, and N Mehlmer. 2023. Biosynthetic potential of Hypogymnia holobionts: Insights into secondary metabolite pathways. Journal of Fungi, 9: 546. https://doi.org/10.3390/jof9050546

+ Ding Y, Zhao L, Wang G, Shi Y, Guo G, Liu C, Chen Z, Coker OO, She J, and J Yu. 2023. PacBio sequencing of human fecal samples uncovers the DNA methylation landscape of 22 673 gut phages. Nucleic Acids Research, gkad977. https://doi.org/10.1093/nar/gkad977

+ Haro-Moreno JM, Cabello-Yeves PJ, Garcillan-Barcia MP, Zakharenko A, Zemskaya TI, and F Rodrigue-Valera. 2023. A novel and diverse group of Candidatus Patescibacteria from bathypelagic Lake Baikal revealed through long-read metagenomics. Environmental Microbiome, 18: 12. https://doi.org/10.1186%2Fs40793-023-00473-1

+ Jia L, Wu Y, Dong Y, Chen J, Chen W-H, and X-M Zhao. 2023. A survey on computational strategies for genome-resolved gut metagenomics. Briefings in Bioinformatics, 2023, 1-13. https://doi.org/10.1093/bib/bbad162

+ Jiang F, Li Q, Wang S, Shen T, Wang H, Wang A, Xu D, Yuan L, Lei L, Chen R, Yang B, Deng Y, and W Fan. 2023. Recovery of metagenome-assembled microbial genomes from a full-scale biogas plant of food waste by Pacific Biosciences high-fidelity sequencing. Frontiers in Microbiology, 13: 1095497. https://doi.org/10.3389/fmicb.2022.1095497

+ Merges D, Dal Grande F, Valim H, Singh G, and I Schmitt. 2023. Gene abundance linked to climate zone: parallel evolution of gene content along elevation gradients in lichenized fungi. Frontiers in Microbiology, 14: 1097787. https://doi.org/10.3389/fmicb.2023.1097787

+ Nielsen TK, Forero-Junco LM, Kot W, Moineau S, Hansen LH, and L Riber. 2023. Detection of nucleotide modifications in bacteria and bacteriophages: strengths and limitations of current technologies and software. Molecular Ecology, 32: 1236-1247. https://doi.org/10.1111/mec.16679

+ Orellana LH, Kruger K, Sidhu C, and R Amann. 2023. Comparing genomes recovered from time-series metagenomes using long- and short-read sequencing technologies. Microbiome, 11: 105. https://doi.org/10.1186/s40168-023-01557-3

+ Priest T, von Appen W-J, Oldenburg E, Popa O, Torres-Valdes S, Bienhold C, Metfies K, Boulton W, Mock T, Fuchs BM, Amann R, Boetius A, and M Wietz. 2023. Atlantic water influx and sea-ice cover drive taxonomic and functional shifts in Arctic marine bacterial communities. The ISME Journal, https://doi.org/10.1038/s41396-023-01461-6

+ Saak CC, Pierce EC, Dinh CB, Portik D, Hall R, Ashby M, and RJ Dutton. 2023. Longitudinal, multi-platform metagenomics yields a high-quality genomic catalog and guides an in vitro model for cheese communities. mSystems, 8:e00701-22. https://doi.org/10.1128/msystems.00701-22

+ Schaerer L, Putman L, Bigcraft I, Byrne E, Kulas D, Zolghadr A, Aloba S, Ong R, Shonnard D, and S Techtmann. 2023. Coexistence of specialist and generalist species within mixed plastic derivative-utilizing microbial communities. Microbiome, 11: 224. https://doi.org/10.1186/s40168-023-01645-4

+ Sidhu C, Kirstein IV, Meunier CL, Rick J, Fofonova V, Wiltshire KH, Steinke N, Vidal-Melgosa S, Hehemann J-H, Huettel B, Schweder T, Fuchs BM, Amann RI, and H Teeling. 2023. Dissolved storage glycans shaped the community composition of abundant bacterioplankton clades during a North Sea spring phytoplankton bloom. Microbiome, 11: 77. https://doi.org/10.1186/s40168-023-01517-x

+ Tao Y, Xun F, Zhao C, Mao Z, Li B, Xing P, and QL Wu. 2023. Improved assembly of metagenome-assembled genomes and viruses in Tibetan saline lake sediment by HiFi metagenomic sequencing. Microbiology Spectrum, 11: e03328-22. https://doi.org/10.1128/spectrum.03328-22 

+ Zhang Z, Yang C, Veldsman WP, Fang X, and L Zhang. 2023. Benchmarking genome assembly methods on metagenomic sequencing data. Briefings in Bioinformatics, 24: 1-17. https://doi.org/10.1093/bib/bbad087

+ Zhang Z-F, Liu L-R, Pan Y-P, Pan J, and M Li. 2023. Long-read assembled metagenomic approaches improve our understanding on metabolic potentials of microbial community in mangrove sediments. Microbiome, 11: 188. https://doi.org/10.1186/s40168-023-01630-x

## 2022

+ Adler A, Poirier S, Pagni M, Maillard J, and C Holliger. 2022. Disentangle genus microdiversity within a complex microbial community by using a multi-distance long-read binning method: example of Candidatus Accumulibacter. Environmental Microbiology, 24: 2136-2156. https://doi.org/10.1111/1462-2920.15947

+ Antipov D, Rayko M, Kolmogorov M, and PA Pevzner. 2022. viralFlye: assembling viruses and identifying their hosts from long-read metagenomics data. Genome Biology, 23: 57. https://doi.org/10.1186/s13059-021-02566-x

+ Bickhart DM, Kolmogorov M, Tseng E, Portik DM, Korobeynikov A, Tolstoganov I, Uritskiy G, Liachko I, Sullivan ST, Shin SB, Zorea A, Andreu VP, Panke-Buisse K, Medema MH, Mizrahi I, Pevzner PA, and TPL Smith. 2022. Generating lineage-resolved, complete metagenome-assembled genomes from complex microbial communities. Nature Biotechnology, 40: 711-719. https://doi.org/10.1038/s41587-021-01130-z

+ Feng X, Cheng H, Portik D, and H Li. 2022. Metagenome assembly of high-fidelity long reads with hifiasm-meta. Nature Methods, 19: 671–674. https://doi.org/10.1038/s41592-022-01478-3 

+ Gehrig JL, Portik DM, Driscoll MD, Jackson E, Chakraborty S, Gratalo D, Ashby M, and R Valladares. 2022. Finding the right fit: evaluation of short-read and long-read sequencing approaches to maximize the utility of clinical microbiome data. Microbial Genomics, 8: 000794. https://doi.org/10.1099/mgen.0.000794

+ Jin H, You L, Zhao F, Li S, Ma T, Kwok L-Y, Xu H, and Z Sun. 2022. Hybrid, ultra-deep metagenomic sequencing enables genomic and functional characterization  of low-abundance species in the human gut microbiome. Gut Microbiomes, 14: e 2021790. https://doi.org/10.1080/19490976.2021.2021790

+ Kato S, Masuda S, Shibata A, Shirasu K, and M Ohkuma. 2022. Insights into ecological roles of uncultivated bacteria in Katase hot spring sediment from long-read metagenomics. Frontiers in Microbiology, 13: 1045931. https://doi.org/10.3389/fmicb.2022.1045931

+ Kim CY, Ma J, and I Lee. 2022. HiFi metagenomic sequencing enables assembly of accurate and complete genomes from human gut microbiota. Nature Communications, 13: 6367. https://doi.org/10.1038/s41467-022-34149-0

+ Luo X, Kang X, and A Schonhuth. 2022. Enhancing long-read-based strain-aware metagenome assembly. Frontiers in Genetics, 13: 868280. https://doi.org/10.3389/fgene.2022.868280

+ Meslier V, Quinquis B, Da Silva K, Plaza Onate F, Pons N, Roume H, Podar M, and M Almeida. 2022. Benchmarking second and third-generation sequencing platforms for microbial metagenomics. Scientific Data, 9: 694. https://doi.org/10.1038/s41597-022-01762-z

+ Noonan AJC, Qiu Y, Kieft B, Formby S, Liu T, Dofher K, Koch M, and SJ Hallam. 2022. Metagenome-assembled genomes for “Candidatus Phormidium sp. Strain AB48” and co-occurring microorganisms from an industrial photobioreactor environment. Microbiology Resource Announcements, e0044722. https://doi.org/10.1128/mra.00447-22

+ Patin NV, and KD Goodwin. 2022. Long-read sequencing improves recovery of picoeukaryotic genomes and zooplankton marker genes from marine metagenomes. mSystems, 7: e00595-22. https://doi.org/10.1128/msystems.00595-22

+ Petrone JR, Munoz-Beristain A, Glusberger PR, Russell JT, and EW Triplett. 2022. Unamplified, long-read metagenomic sequencing approach to close endosymbiotic genomes of low-biomass insect populations. Microorganisms, 10: 513. https://doi.org/10.3390/microorganisms10030513 

+ Plaza Onate F, Roume H, and M Almeida. 2022. Recovery of metagenome-assembled genomes from a human fecal sample with Pacific Biosciences high-fidelity sequencing. Microbiology Resource Announcements, 11: e0025022. https://doi.org/10.1128/mra.00250-22

+ Portik DM, Brown CT, and NT Pierce-Ward. 2022. Evaluation of taxonomic classification and profiling methods for long-read shotgun metagenomic sequencing datasets. BMC Bioinformatics, 23: 541. https://doi.org/10.1186/s12859-022-05103-0

+ Seong HJ, Roux S, Hwang CY, and WJ Sul. 2022. Marine DNA methylation patterns are associated with microbial community composition and inform virus-host dynamics. Microbiome, 10: 157. https://doi.org/10.1186/s40168-022-01340-w

+ Sereika M, Kirkegaard RH, Karst SM, Michaelsen TY, Sorensen EA, Wollenberg RD, and M Albertsen. 2022. Oxford Nanopore R10.4 long-read sequencing enables the generation of near-finished bacterial genomes from pure cultures and metagenomes without short-read or reference polishing. Nature Methods, 19: 823-826. https://doi.org/10.1038/s41592-022-01539-7

+ Slizovskiy IB, Oliva M, Settle JK, Zyskina LV, Prosperi M, Boucher C, and NR Noyes. 2022. Target-enriched long-read sequencing (TELSeq) contextualizes antimicrobial resistance genes in metagenomes. Microbiome, 10: 185. https://doi.org/10.1186/s40168-022-01368-y

+ Wickramarachchi A, and Y Lin. 2022. Binning long reads in metagenomics datasets using composition and coverage information. Algorithms for Molecular Biology, 17: 14. https://doi.org/10.1186/s13015-022-00221-z

+ Wilbanks EG, Dore H, Ashby MH, Heiner C, Roberts RJ, and JA Eisen. 2022. Metagenomic methylation patterns resolve bacterial genomes of unusual size and structural complexity. The ISME Journal, 16: 1921-1931.  https://doi.org/10.1038/s41396-022-01242-7

+ Xu G, Zhang L, Liu X, Guan F, Xu Y, Yue H, Huang J-Q, Chen J, Wu N, and J Tian. 2022. Combined assembly of long and short sequencing reads improve the efficiency of exploring the soil metagenome. BMC Genomics, 23: 37. https://doi.org/10.1186/s12864-021-08260-3

+ Zaragoza-Solas A, Haro-Moreno JM, Rodriguez-Valera F, and M Lopez-Perez. 2022. Long-read metagenomics improves the recovery of viral diversity from complex natural marine samples. mSystems, 7, e0019222. https://doi.org/10.1128/msystems.00192-22

+ Zhang Y, Jiang F, Yang B, Wang S, Wang H, Wang A, Xu D, and W Fan. 2022. Improved microbial genomes and gene catalog of the chicken gut from metagenomic sequencing of high-fidelity long reads. GigaScience, 11: 1-12. https://doi.org/10.1093/gigascience/giac116

+ Zhao L, Shi Y, Lau HC-H, Liu W, Luo G, Wang G, Liu C, Pan Y, Zhou Q, Ding Y, Sung JJ-Y, and J Yu. 2022. Uncovering 1058 novel human enteric DNA viruses through deep long-read third-generation sequencing and their clinical impact. Gastroenterology, 163: 699-711. https://doi.org/10.1053/j.gastro.2022.05.048


## 2021

+ Haro-Moreno JM, Lopez-Perez M, and F Rodriguez-Valera. 2021. Enhanced recovery of microbial genes and genomes from a marine water column using long-read metagenomics. Frontiers in Microbiology, 12: 708782. https://doi.org/10.3389/fmicb.2021.708782

+ Kiguchi Y, Nishijima S, Kumar N, Hattori M, and W Suda. 2021. Long-read metagenomics of multiple displacement amplified DNA of low-biomass human gut phageomes by SACRA pre-processing chimeric reads. DNA Research, 28: dsab019. https://doi.org/10.1093/dnares/dsab019

+ Priest T, Orellana LH, Huettel B, Fuchs BM, and R Amann. 2021. Microbial metagenome-assembled genomes of the Fram Strait from short and long read sequencing platforms. PeerJ, 9: e11721. https://doi.org/10.7717/peerj.11721

## 2020

+ Holm JB, France MT, Ma B, McComb E, Robinson CK, Mehta A, Tallon LJ, Brotman RM, and J Ravel. 2020. Comparative metagenome-assembled genome analysis of "Candidatus Lachnocurva vaginae", formerly known as bacterial vaginosis-associated bacterium-1 (BVAB1). Frontiers in Cellular and Infection Microbiology, 10: 117. https://doi.org/10.3389/fcimb.2020.00117

+ Xie H, Yang C, Sun Y, Igarashi Y, Jin T, and F Luo. 2020. PacBio Long Reads Improve Metagenomic Assemblies, Gene Catalogs, and Genome Binning. Frontiers in Genetics, 11: 516269. https://doi.org/10.3389/fgene.2020.516269

## 2019

+ Bickhart DM, Watson M, Koren S, Panke-Buisse K, Cersosimo LM, Press MO, Van Tassell CP, Van Kessel JAS, Haley BJ, Kim SW, Heiner C, Suen G, Bakshy K, Liachko I, Sullivan ST, Myer PR, Ghurye J, Pop M, Weimer PJ, Phillippy AM, and TPL Smith. 2019. Assignment of virus and antimicrobial resistance genes to microbial hosts in a complex microbial community by combined long-read assembly and proximity ligation. Genome Biology, 20, 153. https://doi.org/10.1186/s13059-019-1760-x

+ Suzuki Y, Nishijima S, Furuta Y, Yoshimura J, Suda W, Oshima K, Hattori M, and S Morishita. 2019. Long-read metagenomic exploration of extrachromosomal mobile genetic elements in the human gut. Microbiome, 7: 119. https://doi.org/10.1186/s40168-019-0737-z

## 2018

+ Liu Y-R, Johs A, Bi L, Lu X, Hu H-W, Sun D, He J-Z, and B Gu. 2018. Unraveling microbial communities associated with methylmercury production in paddy soils. Environmental Science and Technology, 52: 13110-13118. https://doi.org/10.1021/acs.est.8b03052

## 2017

+ Driscoll CB, Otten TG, Brown NM, and TW Dreher. 2017. Towards long-read metagenomics: complete assembly of three novel genomes from bacteria dependent on a diazotrophic cyanobacterium in a freshwater lake co-culture. Standards in Genomic Sciences, 12: 9. https://doi.org/10.1186/s40793-017-0224-8

## 2016

+ Frank JA, Pan Y, Tooming-Klunderud A, Eijsink VGH, McHardy AC, Nederbragt AJ, and PB Pope. 2016. Improved metagenome assemblies and taxonomic binning using long-read circular consensus sequence data. Scientific Reports, 6: 25373. https://doi.org/10.1038%2Fsrep25373

