# HiFi Metagenomics Publications

(Updated 12/24)

## Pre-prints 

+ Fan Y, Ni M, Aggarwala V, Mead EA, Ksiezarek M, Cao L, Kamm MA, Borody T, Paramsothy S, Kaakoush NO, Grinspan A, Faith JJ, and G Fang. 2024. LongTrack: long read metagenomics-based precise tracking of bacterial strains and their genomic changes after fecal microbiota transplantation. bioRxiv, https://doi.org/10.1101/2024.09.30.615906

+ Kutuzova S, Lindez PP, Nielsen KN, Olsen NS, Riber L, Gobbi A, Junco LMF, Dougherty PE, Westergaard JC, Christensen S, Hansen LH, Nielsen M, Nissen JN, and S Rasmussen. 2024. Binning meets taxonomy: TaxVAMB improves metagenome binning using bi-modal variational autoencoder. bioRxiv, https://doi.org/10.1101/2024.10.25.620172

+ Mainguy J, Vienne M, Fourquet J, Darbot V, Noirot C, Castinel A, Combes S, Gaspin C, Milan D, Donnadieu C, Iampietro C, Bouchez O, Pascal G, and C Hoede. 2024. metaWGS, a comprehensive workflow to analyze metagenomic data using Illumina or PacBio HiFi reads. bioRxiv, https://doi.org/10.1101/2024.09.13.612854

+ Mao Y, Shisler JL, and TH Nguyen. 2024. Enhanced detection for antibiotic resistance genes in wastewater using CRISPR-enriched metagenomic method. bioRxiv, https://doi.org/10.1101/2024.07.30.605462

+ Martayan I, Robidou L, Shibuya Y, and A Limasset. 2024. Hyper-k-mers: efficient streaming k-mers representation. bioRxiv, https://doi.org/10.1101/2024.11.06.620789

+ Minich JJ, Allsing N, Din MO, Tisza MJ, Maleta K, McDonald D, Hartwick N, Mamerto A, Brennan C, Hansen L, Shaffer J, Murray ER, Duong T, Knight R, Stephenson K, Manary MJ, and TP Michael. 2024. Culture-independent meta-pangenomics enabled by long-read metagenomics reveals novel associations with pediatric undernutrition. Cell, (sneak peak) http://dx.doi.org/10.2139/ssrn.4960739

+ Portik DM, Feng X, Benoit G, Nasko DJ, Auch B, Bryson SJ, Cano R, Carlin M, Damerum A, Farthing B, Grove JR, Islam M, Langford KW, Liachko I, Locken K, Mangelson H, Tang S, Zhang S, Quince C, and JE Wilkinson. 2024. Highly accurate metagenome-assembled genomes from human gut microbiota using long-read assembly, binning, and consolidation methods. bioRxiv, https://doi.org/10.1101/2024.05.10.593587

+ Priest T, Oldenburg E, Popa O, Dede B, Metfies K, von Appen W-J, Torres-Valdes S, Bienhold C, Fuchs BM, Amann R, Boetius A, and M Wietz. 2024. Seasonal recurrence and modular assembly of an Arctic pelagic marine microbiome. bioRxiv, https://doi.org/10.1101/2024.05.10.593482

+ Sapoval N, Liu Y, Curry K, Kille B, Huang W, Kokroko N, Nute MG, Tyshaieva A, Dilthey AT, Molloy E, and TJ Treangan. 2024. Lightweight taxonomic profiling of long-read sequenced metagenomes with Lemur and Magnet. bioRxiv, https://doi.org/10.1101/2024.06.01.596961

+ Shaw J, Boucher C, Yu YW, Noyes N, and H Li. 2024. devider: long-read reconstruction of many diverse haplotypes. bioRxiv, https://doi.org/10.1101/2024.11.05.621838

+ Song MJ, Li F-W, Freund F, Tribble CM, Toffelmier E, Miller C, Shaffer HB, and CJ Rothfels. 2024. The nitrogen-fixing fern Azolla has a complex microbiome characterized by multiple modes of transmission. bioRxiv, https://doi.org/10.1101/2024.05.20.592813

+ Suarez-Moo P, Haro-Moreno JM, and F Rodriguez-Valera. 2024. Microdiversity in marine pelagic ammonia-oxidizing archaeal populations. bioRxiv, https://doi.org/10.1101/2024.04.23.590705

+ Viver T, Knittel K, Amann R, and LH Orellana. 2024. Deep long-read metagenomic sequencing reveals niche differentiation in carbon cycling potential between benthic and planktonic microbial populations. bioRxiv, https://doi.org/10.1101/2024.06.04.597336

+ Wang G, Zhao L, Shi Y, Qu F, Ding Y, Liu W, Liu C, Luo G, Li M, Bai X, Li L, Ho Y-P, and J Yu. 2023. High-throughput generic single-entity sequencing using droplet microfluidics. bioRxiv, https://doi.org/10.1101/2023.08.15.549386

+ Wang H, Sun C, Li Y, Chen J, Zhao X-M, and W-H Chen. 2024. Complementary insights into gut viral genomes: a comparative benchmark of short- and long-read metagenomes using diverse assemblers and binners. Research Square, 
https://doi.org/10.21203/rs.3.rs-5088576/v1

+ West-Roberts J, Valentin-Alvarado L, Mullen S, Sachdeva R, Smith J, Hug LA, Gregoire DS, Liu W, Lin T-Y, Husain G, Amano Y, Ly L, and JF Banfield. 2023. Giant genes are rare but implicated in cell wall degradation by predatory bacteria. bioRxiv, https://doi.org/10.1101/2023.11.21.568195

+ Yamamura T, Takewaki D, Kiguchi Y, Masuoka H, Manu M, Raveney B, Narushima S, Kurokawa R, Ogata Y, Kimura Y, Sato N, Ozawa Y, Yagishita S, Araki T, Miyake S, Sato W, and W Suda. 2023. Horizontal gene transfer shapes pathogenic bacteria in multiple sclerosis. Research Square, https://dx.doi.org/10.21203/rs.3.rs-3716024/v1

+ Zhang S, Li Q, Peng Y, Huo J, Ran T, Zhang X, Wang R, Jiao J, Jiang A, Luo G, Zhang Z, Qiu Q, Li Z, Mao S, Yu Z, Tan Z, Dong X, and M Wang. 2024. Spatial heterogeneity of viral communities across the gastrointestinal tracts of ruminants. Research Square, https://www.researchsquare.com/article/rs-4244627/v1

+ Zhang W, Liu Y, Xu J, Chen E, Schonhuth A, and X Luo. 2024. PanTax: strain-level taxonomic classification of metagenomic data using pangenome graphs. bioRxiv: https://doi.org/10.1101/2024.11.15.623887


## 2024

+ Agustinho DP, Fu Y, Menon VK, Metcalf GA, Treangan TJ, and FJ Sedlazeck. 2024. Unveiling microbial diversity: harnessing long-read sequencing technology. Nature Methods, https://doi.org/10.1038/s41592-024-02262-1

+ Aldeguer-Riquelme B, Conrad RE, Anton J, Rossello-Mora R, and KT Konstantinidis. 2024. A natural ANI gap that can define intra-species units of bacteriophages and other viruses. mBio, 8: e01536-24. https://doi.org/10.1128/mbio.01536-24

+ Benoit G, Raguideau S, James R, Phillippy AM, Chikhi R, and C Quince. 2024. High-quality metagenome assembly from long accurate reads with metaMDBG. Nature Biotechnology, https://doi.org/10.1038/s41587-023-01983-6

+ Bulka O, and EA Edwards. 2024. Metagenomic sequences from anaerobic chloroform and dichloromethane degrading microbial communities. Microbiology Resource Announcements, e00391-24. https://doi.org/10.1128/mra.00391-24

+ Bulka O, and EA Edwards. 2024. Two distinct Dehalobacter metagenome-assembled genomes from anaerobic chloroform and dichloromethane degrading consortia. Microbiology Resource Announcements, e00803-24. https://doi.org/10.1128/mra.00803-24

+ Bulka O, and EA Edwards. 2024. Metagenome from ACT-3/CF: an anaerobic chloroform-degrading microbial community. Microbiology Resource Announcements, 13: e00674-24. https://doi.org/10.1128/mra.00674-24

+ Bulka O, Picott K, Mahadevan R, and EA Edwards. 2024. From mec cassette to rdhA: a key Dehalobacter genomic neighborhood in a chloroform and dichloromethane-transforming microbial consortium. Applied and Environmental Microbiology, 90: e0073224. https://doi.org/10.1128/aem.00732-24

+ Chen J, Sun C, Dong Y, Jin M, Lai S, Jia L, Zhao X, Wang H, Gao NL, Bork P, Liu Z, Chen W-H, and X-M Zhao. 2024. Efficient recovery of complete gut phage genomes by combined short- and long-sequencing. Advanced Science, 2305818. https://doi.org/10.1002/advs.202305818

+ Chen X, Yin X, Shi X, Yan W, Yang Y, Liu L, and T Zhang. 2023. Melon: metagenomic long-read-based taxonomic identification and quantification using marker genes. Genome Biology, 25: 226. https://doi.org/10.1186/s13059-024-03363-y

+ Cook R, Brown N, Rihtman B, Michniewski S, Redgwell T, Clokie M, Stekel DJ, Chen Y, Scanlan DJ, Hobman JL, Nelson A, Jones MA, Smith D, and A Millard. 2024. The long and short of it: benchmarking viromics using Illumina, Nanopore, and PacBio sequencing technologies. Microbial Genomics, 10: 001198. https://doi.org/10.1099/mgen.0.001198

+ Dong W, Fan X, Guo Y, Wang S, Jia S, Lv N, Yuan T, Pan Y, Xue Y, Chen X, Xiong Q, Yang R, Zhao W, and B Zhu. 2024. An expanded database and analytical toolkit for identifying bacterial virulence factors and their associations with chronic diseases. Nature Communications, 15: 8084. https://doi.org/10.1038/s41467-024-51864-y

+ Eisenhofer R, Nesme J, Santos-Bay L, Koziol A, Sorenson SJ, Alberdi A, and O Aizpurua. 2024. A comparison of short-read, HiFi long-read and hybrid strategies for genome-resolved metagenomics. Microbiology Spectrum, 12: e03590-23. https://doi.org/10.1128/spectrum.03590-23

+ Feng X, and H Li. 2024. Evaluating and improving the representation of bacterial contents in long-read metagenome assemblies. Genome Biology, 25: 92. https://doi.org/10.1186/s13059-024-03234-6

+ Funnicelli MIG, de Carvalho LAL, Teheran-Sierra LG, Dibelli SC, de Macedo Lemos EG, and DG Pinheiro. 2024. Unveiling genomic features linked to traits of plant growth-promoting bacterial communities from sugarcane. Science of the Total Environment, 947: 174577. https://doi.org/10.1016/j.scitotenv.2024.174577

+ Glendinning L, Wu Z, Vervelde L, Watson M, and A Balic. 2024. Infectious bronchitis virus vaccination, but not the presence of XCRI, is correlated with large differences in chicken caecal microbiota. Microbial Genomics, 10: 001289.  https://doi.org/10.1099/mgen.0.001289

+ Han Y, He J, Li M, Peng Y, Jiang H, Zhao J, Li Y, and F Deng. 2024. Unlocking the potential of metagenomics with the PacBio high-fidelity sequencing technology. Microorganisms, 12: 2482. https://doi.org/10.3390/microorganisms12122482

+ Hui X, Yang J, Sun J, Liu F, and W Pan. 2024. MCSS: microbial community simulator based on structure. Frontiers in Microbiology, 15: 1358257. https://doi.org/10.3389/fmicb.2024.1358257

+ Kazantseva E, Donmez A, Frolova M, Pop M, and M Kolmogorov. 2024. Strainy: phasing and assembly of strain haplotypes from long-read metagenome sequencing. Nature Methods, https://doi.org/10.1038/s41592-024-02424-1

+ Kutuzova S, Nielsen M, Piera P, Nissen JN, and S Rasmussen. 2024. Taxometer: improving taxonomic classification of metagenomic contigs. Nature Communications, 15: 8357. https://doi.org/10.1038/s41467-024-52771-y

+ Lai S, Wang H, Bork P, Chen W-H, and X-M Zhao. 2024. Long-read sequencing reveals extensive gut phageome structural variations driven by genetic exchange with bacterial hosts. Science Advances, 10: eadn3316. https://doi.org/10.1126/sciadv.adn3316

+ Manning VA, Moore PA, and KM Trippe. 2024. Metagenome-assembled genomes of an acid-tolerant nitrifying bacterial community isolated from a bioreactor used in ammonium scrubbers at animal-rearing facilities. Microbiology Resource Announcements, e00386-24. https://doi.org/10.1128/mra.00386-24

+ Maric J, Krizanovic K, Riondet S, Nagarajan N, and M Sikic. 2024. Comparative analysis of metagenomic classifiers for long-read sequencing datasets. BMC Bioinformatics, 25: 15. https://doi.org/10.1186/s12859-024-05634-8

+ Masuda S, Gan P, Kiguchi Y, Anda M, Sasaki K, Shibata A, Iwasaki W, Suda W, and K Shirasu. 2024. Uncovering microbiomes of the rice phyllosphere using long-read metagenomic sequencing. Communications Biology, 7: 357. https://doi.org/10.1038/s42003-024-05998-w

+ Myeong NR, Choe Y-H, Shin SC, Kim J, Sul WJ, and M Kim. 2024. Genomic profiling of Antarctic geothermal microbiomes using long-read, Hi-C, and single-cell techniques. Scientific Data, 11: 1023. https://doi.org/10.1038/s41597-024-03875-z

+ Nesbo CL, Yang MI, Sharan AA, Meyer T, and EE Edwards. 2024. Metagenomes and metagenome assembled genomes from anaerobic digesters at three Canadian pulp and paper mills. Microbiology Resource Announcements, e00561-24. https://doi.org/10.1128/mra.00561-24

+ Peres de Silva R, Suphavilai C, and N Nagrajan. 2024. MetageNN: a memory-efficient neural network taxonomic classifier robust to sequencing errors and missing genomes. BMC Bioinformatics, 25: 153. https://doi.org/10.1186/s12859-024-05760-3

+ Richy E, Dobbler PT, Tlaskal V, Lopez-Mondejar R, Baldrian P, and M Kyselkova. 2024. PacBio HiFi sequencing sheds light on key bacteria contributing to deadwood decomposition processes. Environmental Microbiome, 19: 99. https://doi.org/10.1186/s40793-024-00639-5

+ Saini JS, Adler A, Cardona L, Rodilla Ramirez PN, Pei R, and C Holliger. 2024. Microbial genome collection of aerobic granular sludge cultivated in sequencing batch reactors using different carbon source mixtures. Microbiology Resource Announcements, 13: e00102-24. https://doi.org/10.1128/mra.00102-24

+ Sakurai R, Fukuda Y, and C Tada. 2024. Circular metagenome-assembled genome of Candidatus Patescibacteria recovered from anaerobic digestion sludge. Microbiology Resource Announcements, 13: e00083-24. https://doi.org/10.1128/mra.00083-24

+ Shaw J, and YW Yu. 2024. Rapid species-level metagenome profiling and containment estimation with sylph. Nature Biotechnology, https://doi.org/10.1038/s41587-024-02412-y

+ Shaw J, Gounot J-S, Chen H, Nagarajan N, and YW Yu. 2024. Floria: fast and accurate strain haplotyping in metagenomes. Bioinformatics, 40: i30-i38. https://doi.org/10.1093/bioinformatics/btae252

+ Silva-Solar S, Viver T, Wang Y, Orellana LH, Knittel K, and R Amann. 2024. Acidimicrobiia, the actinomycetota of coastal marine sediments: abundance, taxonomy and genomic potential. Systematic and Applied Microbiology, 47: 126555. https://doi.org/10.1016/j.syapm.2024.126555

+ Stewart RD, Myers KS, Amstadt C, Seib M, McMahon KD, and DR Noguera. 2024. Refinement of the “Candidatus Accumulibacter” genus based on metagenomic analysis of biological nutrient removal (BNR) pilot-scale plants operated with reduced aeration. mSystems, 9: e01188-23. https://doi.org/10.1128/msystems.01188-23

+ Suarez-Moo P, Haro-Moreno JM, and F Rodriguez-Valera. 2024. Microdiversity in marine pelagic ammonia-oxidizing archaeal populations in a Mediterranean long-read metagenome. Environmental Microbiology, 26: e16684. https://doi.org/10.1111/1462-2920.16684

+ Takewaki D, Kiguchi Y, Masuoka H, Manu MS, Raveney BJE, Narushima S, Kurokawa R, Ogata Y, Kimura Y, Sato N, Ozawa Y, Yagishita S, Araki T, Miyake S, Sato W, Suda W, and T Yamamura. 2024. Tyzzerella nexilis strains enriched in mobile genetic elements are involved in progressive multiple sclerosis. Cell Reports, 114785. https://doi.org/10.1016/j.celrep.2024.114785

+ Valentin-Alvarado LE, Shi L-D, Appler KE,  Crits-Christoph A, De Anda V, Adler BA, Cui ML, Ly L, Leao P, Roberts RJ, Sachdeva R, Baker BJ, Savage DF, and JF Banfield. 2024. Complete genomes of Asgard archaea reveal diverse integrated and mobile genetic elements. Genome Research, 34: 1-15. https://doi.org/10.1101/gr.279480.124

+ Valentin-Alvarado LE, Appler KE, De Anda V, Schoelmerich MC, West-Roberts J, Kivenson V, Crits-Christoph A, Ly L, Sachdeva R, Greening C, Savage DF, Baker BJ, and JF Banfield. 2024. Asgard archaea modulate potential methanogenesis substrates in wetland soil. Nature Communications, 15: 6384. https://doi.org/10.1038/s41467-024-49872-z

+ Wang Y-C, Mao Y, Fu H-M, Wang J, Weng X, Liu Z-H, Xu X-W, Yan P, Fang F, Guo J-S, Shen Y, and Y-P Chen. 2024. New insights into functional divergence and adaptive evolution of uncultured bacteria in anammox community by complete genome-centric analysis. Science of the Total Environment, 924: 171530. https://doi.org/10.1016/j.scitotenv.2024.171530

+ Xiao JZ, Nesbo CL, Molenda O, Toth CRA, and EA Edwards. 2024. Metagenomic and genomic sequences from a nitrate-reducing benzene-degrading enrichment culture. Microbiology Resource Announcements, e00294-24. https://doi.org/10.1128/mra.00294-24

+ Yadav A, and S Subramanian. 2024. HiFiBGC: an ensemble approach for improved biosynthetic gene cluster detection in PacBio HiFi-read metagenomes. BMC Genomics, 15: 1096. https://doi.org/10.1186/s12864-024-10950-7 

+ Yu W, Luo H, Yang J, Zhang S, Jiang H, Zhao X, Hui X, Sun D, Li L, Wei X-Q, Lonardi S, and W Pan. 2024. Comprehensive assessment of 11 de novo HiFi assemblers on complex eukaryotic genomes and metagenomes. Genome Research, 34: 326-340. https://www.genome.org/cgi/doi/10.1101/gr.278232.123


## 2023

+ Ahmad N, Ritz M, Calchera A, Otte J, Schmitt I, Brueck T, and N Mehlmer. 2023. Biosynthetic potential of Hypogymnia holobionts: Insights into secondary metabolite pathways. Journal of Fungi, 9: 546. https://doi.org/10.3390/jof9050546

+ Chen X, Molenda O, Brown CT, Toth CRA, Guo S, Luo F, Howe J, Nesbo CL, He C, Montabana EA, Cate JHD, Banfield JF, and EA Edwards. 2023. “Candidatus Nealsonbacteria” are likely biomass recycling ectosymbionts of methanogenic Archaea in a stable benzene-degrading enrichment culture. Applied and Environmental Microbiology, 89: e00025-23. https://doi.org/10.1128/aem.00025-23

+ Ding Y, Zhao L, Wang G, Shi Y, Guo G, Liu C, Chen Z, Coker OO, She J, and J Yu. 2023. PacBio sequencing of human fecal samples uncovers the DNA methylation landscape of 22 673 gut phages. Nucleic Acids Research, gkad977. https://doi.org/10.1093/nar/gkad977

+ Haro-Moreno JM, Cabello-Yeves PJ, Garcillan-Barcia MP, Zakharenko A, Zemskaya TI, and F Rodrigue-Valera. 2023. A novel and diverse group of Candidatus Patescibacteria from bathypelagic Lake Baikal revealed through long-read metagenomics. Environmental Microbiome, 18: 12. https://doi.org/10.1186%2Fs40793-023-00473-1

+ Jang YJ, Kang J-S, and EH Bae. 2023. Metagenome-assembled genomes of the GU0601 sample (the Han River, South Korea). Microbiology Resource Announcements, 12: e00688-23. https://doi.org/10.1128/MRA.00688-23

+ Jia L, Wu Y, Dong Y, Chen J, Chen W-H, and X-M Zhao. 2023. A survey on computational strategies for genome-resolved gut metagenomics. Briefings in Bioinformatics, 2023, 1-13. https://doi.org/10.1093/bib/bbad162

+ Jiang F, Li Q, Wang S, Shen T, Wang H, Wang A, Xu D, Yuan L, Lei L, Chen R, Yang B, Deng Y, and W Fan. 2023. Recovery of metagenome-assembled microbial genomes from a full-scale biogas plant of food waste by Pacific Biosciences high-fidelity sequencing. Frontiers in Microbiology, 13: 1095497. https://doi.org/10.3389/fmicb.2022.1095497

+ Merges D, Dal Grande F, Valim H, Singh G, and I Schmitt. 2023. Gene abundance linked to climate zone: parallel evolution of gene content along elevation gradients in lichenized fungi. Frontiers in Microbiology, 14: 1097787. https://doi.org/10.3389/fmicb.2023.1097787

+ Nesbo CL, Fitamo TM, Lee H, and EA Edwards. 2023. Metagenomes and metagenome-assembled genomes from a sequentially fed anaerobic digester treating solid organic municipal waste. Microbiology Resource Announcements, 13: e00919-23. https://doi.org/10.1128/mra.00919-23 

+ Nielsen TK, Forero-Junco LM, Kot W, Moineau S, Hansen LH, and L Riber. 2023. Detection of nucleotide modifications in bacteria and bacteriophages: strengths and limitations of current technologies and software. Molecular Ecology, 32: 1236-1247. https://doi.org/10.1111/mec.16679

+ Orellana LH, Kruger K, Sidhu C, and R Amann. 2023. Comparing genomes recovered from time-series metagenomes using long- and short-read sequencing technologies. Microbiome, 11: 105. https://doi.org/10.1186/s40168-023-01557-3

+ Priest T, von Appen W-J, Oldenburg E, Popa O, Torres-Valdes S, Bienhold C, Metfies K, Boulton W, Mock T, Fuchs BM, Amann R, Boetius A, and M Wietz. 2023. Atlantic water influx and sea-ice cover drive taxonomic and functional shifts in Arctic marine bacterial communities. The ISME Journal, https://doi.org/10.1038/s41396-023-01461-6

+ Priest T, Vidal-Melgosa S, Hehemann J-H, Amann R, and BM Fuchs. 2023. Carbohydrates and carbohydrate degradation gene abundance and transcription in Atlantic waters of the Arctic. ISME Communications, 3: 130. https://doi.org/10.1038/s43705-023-00324-7

+ Rodriguez Ruiz A, and AR Van Dam. 2023. Metagenomic binning of PacBio HiFi data prior to assembly reveals a complete genome of Cosmopolites sordidus (Germar) (Coleopterea: Curculionidae, Dryophthorinae) the most damaging arthropod pest of bananas and plantains. PeerJ 11: e16276. https://doi.org/10.7717/peerj.16276

+ Saak CC, Pierce EC, Dinh CB, Portik D, Hall R, Ashby M, and RJ Dutton. 2023. Longitudinal, multi-platform metagenomics yields a high-quality genomic catalog and guides an in vitro model for cheese communities. mSystems, 8:e00701-22. https://doi.org/10.1128/msystems.00701-22

+ Schaerer L, Putman L, Bigcraft I, Byrne E, Kulas D, Zolghadr A, Aloba S, Ong R, Shonnard D, and S Techtmann. 2023. Coexistence of specialist and generalist species within mixed plastic derivative-utilizing microbial communities. Microbiome, 11: 224. https://doi.org/10.1186/s40168-023-01645-4

+ Sidhu C, Kirstein IV, Meunier CL, Rick J, Fofonova V, Wiltshire KH, Steinke N, Vidal-Melgosa S, Hehemann J-H, Huettel B, Schweder T, Fuchs BM, Amann RI, and H Teeling. 2023. Dissolved storage glycans shaped the community composition of abundant bacterioplankton clades during a North Sea spring phytoplankton bloom. Microbiome, 11: 77. https://doi.org/10.1186/s40168-023-01517-x

+ Sun C, Chen J, Jin M, Zhao X, Li Y, Dong Y, Gao N, Liu Z, Bork P, Zhao X-M, and W-H Chen. 2023. Long-read sequencing reveals extensive DNA methylations in human gut pangenome contributed by prevalently phage-encoded methyltransferases. Advanced Science, 10: 2302159. https://doi.org/10.1002/advs.202302159

+ Tao Y, Xun F, Zhao C, Mao Z, Li B, Xing P, and QL Wu. 2023. Improved assembly of metagenome-assembled genomes and viruses in Tibetan saline lake sediment by HiFi metagenomic sequencing. Microbiology Spectrum, 11: e03328-22. https://doi.org/10.1128/spectrum.03328-22 

+ Toth CRA, Molenda O, Nesbo C, Luo F, Devine C, Guo S, Chen X, EA Edwards. 2023. Metagenomic and genomic sequences from a methanogenic benzene-degrading consortium. Microbiology Resource Announcements, 12: e01342-22. https://doi.org/10.1128/mra.01342-22

+ Zhang Z, Yang C, Veldsman WP, Fang X, and L Zhang. 2023. Benchmarking genome assembly methods on metagenomic sequencing data. Briefings in Bioinformatics, 24: 1-17. https://doi.org/10.1093/bib/bbad087

+ Zhang Z-F, Liu L-R, Pan Y-P, Pan J, and M Li. 2023. Long-read assembled metagenomic approaches improve our understanding on metabolic potentials of microbial community in mangrove sediments. Microbiome, 11: 188. https://doi.org/10.1186/s40168-023-01630-x


## 2022

+ Adler A, Poirier S, Pagni M, Maillard J, and C Holliger. 2022. Disentangle genus microdiversity within a complex microbial community by using a multi-distance long-read binning method: example of Candidatus Accumulibacter. Environmental Microbiology, 24: 2136-2156. https://doi.org/10.1111/1462-2920.15947

+ Antipov D, Rayko M, Kolmogorov M, and PA Pevzner. 2022. viralFlye: assembling viruses and identifying their hosts from long-read metagenomics data. Genome Biology, 23: 57. https://doi.org/10.1186/s13059-021-02566-x

+ Bickhart DM, Kolmogorov M, Tseng E, Portik DM, Korobeynikov A, Tolstoganov I, Uritskiy G, Liachko I, Sullivan ST, Shin SB, Zorea A, Andreu VP, Panke-Buisse K, Medema MH, Mizrahi I, Pevzner PA, and TPL Smith. 2022. Generating lineage-resolved, complete metagenome-assembled genomes from complex microbial communities. Nature Biotechnology, 40: 711-719. https://doi.org/10.1038/s41587-021-01130-z

+ Fedarko MW, Kolmogorov M, and PA Pevzner. 2022. Analyzing rare mutations in metagenomes assembled using long and accurate reads. Genome Research, 32: 2119-2133. https://doi.org/10.1101%2Fgr.276917.122

+ Feng X, Cheng H, Portik D, and H Li. 2022. Metagenome assembly of high-fidelity long reads with hifiasm-meta. Nature Methods, 19: 671–674. https://doi.org/10.1038/s41592-022-01478-3 

+ Fortney NW, Myers KS, Ingle AT, Walters KA, Scarborough MJ, Donohue TJ, and DR Noguera. 2022. Metagenomes and metagenome-assembled genomes from microbiomes metabolizing thin stillage from an ethanol biorefinery. Microbiology Resource Announcements, 11: e00290-22. https://doi.org/10.1128/mra.00290-22 

+ Gehrig JL, Portik DM, Driscoll MD, Jackson E, Chakraborty S, Gratalo D, Ashby M, and R Valladares. 2022. Finding the right fit: evaluation of short-read and long-read sequencing approaches to maximize the utility of clinical microbiome data. Microbial Genomics, 8: 000794. https://doi.org/10.1099/mgen.0.000794

+ Ingle AT, Fortney NW, Myers KS, Walters KA, Scarborough MJ, Donohue TJ, and DR Noguera. 2022. Metagenome-assembled genomes from a microbiome grown in dairy manure hydrolysate. Microbiology Resource Announcements, 11: e00292-22. https://doi.org/10.1128/mra.00292-22

+ Jin H, You L, Zhao F, Li S, Ma T, Kwok L-Y, Xu H, and Z Sun. 2022. Hybrid, ultra-deep metagenomic sequencing enables genomic and functional characterization  of low-abundance species in the human gut microbiome. Gut Microbiomes, 14: e 2021790. https://doi.org/10.1080/19490976.2021.2021790

+ Kato S, Masuda S, Shibata A, Shirasu K, and M Ohkuma. 2022. Insights into ecological roles of uncultivated bacteria in Katase hot spring sediment from long-read metagenomics. Frontiers in Microbiology, 13: 1045931. https://doi.org/10.3389/fmicb.2022.1045931

+ Kim CY, Ma J, and I Lee. 2022. HiFi metagenomic sequencing enables assembly of accurate and complete genomes from human gut microbiota. Nature Communications, 13: 6367. https://doi.org/10.1038/s41467-022-34149-0

+ Loureiro C, Galani A, Gavriilidou A, de Mars, MC, van der Oost J, Medema MH, and D Sipkema. 2022. Comparative metagenomic analysis of biosynthetic diversity across sponge microbiomes highlights metabolic novelty, conservation, and diversification. mSystems, 7: e00357-22. https://doi.org/10.1128/msystems.00357-22

+ Luo X, Kang X, and A Schonhuth. 2022. Enhancing long-read-based strain-aware metagenome assembly. Frontiers in Genetics, 13: 868280. https://doi.org/10.3389/fgene.2022.868280

+ Meslier V, Quinquis B, Da Silva K, Plaza Onate F, Pons N, Roume H, Podar M, and M Almeida. 2022. Benchmarking second and third-generation sequencing platforms for microbial metagenomics. Scientific Data, 9: 694. https://doi.org/10.1038/s41597-022-01762-z

+ Noonan AJC, Qiu Y, Kieft B, Formby S, Liu T, Dofher K, Koch M, and SJ Hallam. 2022. Metagenome-assembled genomes for “Candidatus Phormidium sp. Strain AB48” and co-occurring microorganisms from an industrial photobioreactor environment. Microbiology Resource Announcements, e0044722. https://doi.org/10.1128/mra.00447-22

+ Patin NV, and KD Goodwin. 2022. Long-read sequencing improves recovery of picoeukaryotic genomes and zooplankton marker genes from marine metagenomes. mSystems, 7: e00595-22. https://doi.org/10.1128/msystems.00595-22

+ Petrone JR, Munoz-Beristain A, Glusberger PR, Russell JT, and EW Triplett. 2022. Unamplified, long-read metagenomic sequencing approach to close endosymbiotic genomes of low-biomass insect populations. Microorganisms, 10: 513. https://doi.org/10.3390/microorganisms10030513 

+ Plaza Onate F, Roume H, and M Almeida. 2022. Recovery of metagenome-assembled genomes from a human fecal sample with Pacific Biosciences high-fidelity sequencing. Microbiology Resource Announcements, 11: e0025022. https://doi.org/10.1128/mra.00250-22

+ Portik DM, Brown CT, and NT Pierce-Ward. 2022. Evaluation of taxonomic classification and profiling methods for long-read shotgun metagenomic sequencing datasets. BMC Bioinformatics, 23: 541. https://doi.org/10.1186/s12859-022-05103-0

+ Seong HJ, Roux S, Hwang CY, and WJ Sul. 2022. Marine DNA methylation patterns are associated with microbial community composition and inform virus-host dynamics. Microbiome, 10: 157. https://doi.org/10.1186/s40168-022-01340-w

+ Sereika M, Kirkegaard RH, Karst SM, Michaelsen TY, Sorensen EA, Wollenberg RD, and M Albertsen. 2022. Oxford Nanopore R10.4 long-read sequencing enables the generation of near-finished bacterial genomes from pure cultures and metagenomes without short-read or reference polishing. Nature Methods, 19: 823-826. https://doi.org/10.1038/s41592-022-01539-7

+ Slizovskiy IB, Oliva M, Settle JK, Zyskina LV, Prosperi M, Boucher C, and NR Noyes. 2022. Target-enriched long-read sequencing (TELSeq) contextualizes antimicrobial resistance genes in metagenomes. Microbiome, 10: 185. https://doi.org/10.1186/s40168-022-01368-y

+ Walters KA, Myers KS, Wang H, Fortney NW, Ingle AT, Scarborough MJ, Donohue TJ, and DR Noguera. 2022. Metagenomes and metagenome-assembled genomes from microbial communities fermenting ultrafiltered milk permeate. Microbiology Resource Announcements, 11: e00293-22. https://doi.org/10.1128/mra.00293-22

+ Wickramarachchi A, and Y Lin. 2022. Binning long reads in metagenomics datasets using composition and coverage information. Algorithms for Molecular Biology, 17: 14. https://doi.org/10.1186/s13015-022-00221-z

+ Wilbanks EG, Dore H, Ashby MH, Heiner C, Roberts RJ, and JA Eisen. 2022. Metagenomic methylation patterns resolve bacterial genomes of unusual size and structural complexity. The ISME Journal, 16: 1921-1931.  https://doi.org/10.1038/s41396-022-01242-7

+ Xu G, Zhang L, Liu X, Guan F, Xu Y, Yue H, Huang J-Q, Chen J, Wu N, and J Tian. 2022. Combined assembly of long and short sequencing reads improve the efficiency of exploring the soil metagenome. BMC Genomics, 23: 37. https://doi.org/10.1186/s12864-021-08260-3

+ Zaragoza-Solas A, Haro-Moreno JM, Rodriguez-Valera F, and M Lopez-Perez. 2022. Long-read metagenomics improves the recovery of viral diversity from complex natural marine samples. mSystems, 7, e0019222. https://doi.org/10.1128/msystems.00192-22

+ Zhang Y, Jiang F, Yang B, Wang S, Wang H, Wang A, Xu D, and W Fan. 2022. Improved microbial genomes and gene catalog of the chicken gut from metagenomic sequencing of high-fidelity long reads. GigaScience, 11: 1-12. https://doi.org/10.1093/gigascience/giac116

+ Zhao L, Shi Y, Lau HC-H, Liu W, Luo G, Wang G, Liu C, Pan Y, Zhou Q, Ding Y, Sung JJ-Y, and J Yu. 2022. Uncovering 1058 novel human enteric DNA viruses through deep long-read third-generation sequencing and their clinical impact. Gastroenterology, 163: 699-711. https://doi.org/10.1053/j.gastro.2022.05.048


## 2021

+ Haro-Moreno JM, Lopez-Perez M, and F Rodriguez-Valera. 2021. Enhanced recovery of microbial genes and genomes from a marine water column using long-read metagenomics. Frontiers in Microbiology, 12: 708782. https://doi.org/10.3389/fmicb.2021.708782

+ Kiguchi Y, Nishijima S, Kumar N, Hattori M, and W Suda. 2021. Long-read metagenomics of multiple displacement amplified DNA of low-biomass human gut phageomes by SACRA pre-processing chimeric reads. DNA Research, 28: dsab019. https://doi.org/10.1093/dnares/dsab019

+ Priest T, Orellana LH, Huettel B, Fuchs BM, and R Amann. 2021. Microbial metagenome-assembled genomes of the Fram Strait from short and long read sequencing platforms. PeerJ, 9: e11721. https://doi.org/10.7717/peerj.11721

+ Tedersoo L, Albertsen M, Anslan S, and B Callahan. 2021. Perspectives and benefits of high-throughput long-read sequencing in microbial ecology. Applied and Environmental Microbiology, 87: e00626-21. https://doi.org/10.1128/AEM.00626-21


## 2020

+ Holm JB, France MT, Ma B, McComb E, Robinson CK, Mehta A, Tallon LJ, Brotman RM, and J Ravel. 2020. Comparative metagenome-assembled genome analysis of "Candidatus Lachnocurva vaginae", formerly known as bacterial vaginosis-associated bacterium-1 (BVAB1). Frontiers in Cellular and Infection Microbiology, 10: 117. https://doi.org/10.3389/fcimb.2020.00117

+ Xie H, Yang C, Sun Y, Igarashi Y, Jin T, and F Luo. 2020. PacBio Long Reads Improve Metagenomic Assemblies, Gene Catalogs, and Genome Binning. Frontiers in Genetics, 11: 516269. https://doi.org/10.3389/fgene.2020.516269


## 2019

+ Bickhart DM, Watson M, Koren S, Panke-Buisse K, Cersosimo LM, Press MO, Van Tassell CP, Van Kessel JAS, Haley BJ, Kim SW, Heiner C, Suen G, Bakshy K, Liachko I, Sullivan ST, Myer PR, Ghurye J, Pop M, Weimer PJ, Phillippy AM, and TPL Smith. 2019. Assignment of virus and antimicrobial resistance genes to microbial hosts in a complex microbial community by combined long-read assembly and proximity ligation. Genome Biology, 20, 153. https://doi.org/10.1186/s13059-019-1760-x

+ Sevim V, Lee J, Egan R, Clum A, Hundley H, Lee J, Everroad RC, Detweiler AM, Bebout BM, Pett-Ridge J, Goker M, Murray AE, Lindemann SR, Klenk H-P, O'Malley R, Zane M, Cheng J-F, Copeland A, Daum C, Singer E and T Woyke. 2019. Shotgun metagenome data of a defined mock community using Oxford Nanopore, PacBio, and Illumina technologies. Scientific Data, 6: 285. https://doi.org/10.1038/s41597-019-0287-z


+ Suzuki Y, Nishijima S, Furuta Y, Yoshimura J, Suda W, Oshima K, Hattori M, and S Morishita. 2019. Long-read metagenomic exploration of extrachromosomal mobile genetic elements in the human gut. Microbiome, 7: 119. https://doi.org/10.1186/s40168-019-0737-z


## 2018

+ Liu Y-R, Johs A, Bi L, Lu X, Hu H-W, Sun D, He J-Z, and B Gu. 2018. Unraveling microbial communities associated with methylmercury production in paddy soils. Environmental Science and Technology, 52: 13110-13118. https://doi.org/10.1021/acs.est.8b03052


## 2017

+ Driscoll CB, Otten TG, Brown NM, and TW Dreher. 2017. Towards long-read metagenomics: complete assembly of three novel genomes from bacteria dependent on a diazotrophic cyanobacterium in a freshwater lake co-culture. Standards in Genomic Sciences, 12: 9. https://doi.org/10.1186/s40793-017-0224-8


## 2016

+ Frank JA, Pan Y, Tooming-Klunderud A, Eijsink VGH, McHardy AC, Nederbragt AJ, and PB Pope. 2016. Improved metagenome assemblies and taxonomic binning using long-read circular consensus sequence data. Scientific Reports, 6: 25373. https://doi.org/10.1038%2Fsrep25373

